packages/scripts/src/monitoring-dvm2.0/common.ts

```diff
@@ -0,0 +1,131 @@
+import { VotingTokenEthers, VotingV2Ethers } from "@uma/contracts-node";
+import { TypedEventFilter } from "@uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/commons";
+import { BigNumber } from "ethers";
+import { getContractInstance } from "../utils/contracts";
+import { forkNetwork, getForkChainId, increaseEvmTime } from "../utils/utils";
+
+const hre = require("hardhat");
+const { ethers } = hre;
+
+export const getVotingContracts = async (): Promise<{ votingV2: VotingV2Ethers; votingToken: VotingTokenEthers }> => {
+  if (!process.env.CUSTOM_NODE_URL) throw new Error("CUSTOM_NODE_URL must be defined in env");
+  await forkNetwork(process.env.CUSTOM_NODE_URL);
+  const chainId = await getForkChainId(process.env.CUSTOM_NODE_URL);
+
+  if (!chainId || (chainId != 1 && chainId != 5)) throw new Error("This script should be run on mainnet or goerli");
+
+  const votingV2 = await getContractInstance<VotingV2Ethers>("VotingV2", undefined, chainId);
+  const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken", undefined, chainId);
+  return { votingV2, votingToken };
+};
+
+export const getUniqueVoters = async (votingV2: VotingV2Ethers): Promise<string[]> => {
+  const stakedEvents = await votingV2.queryFilter(votingV2.filters.Staked(null, null, null));
+  const uniqueVoters = new Set<string>(stakedEvents.map((event) => event.args.voter));
+  return Array.from(uniqueVoters);
+};
+
+export const updateTrackers = async (votingV2: VotingV2Ethers, voters: string[]): Promise<void> => {
+  console.log("Updating trackers for all voters");
+  const tx = await votingV2.multicall(
+    voters.map((voter) => votingV2.interface.encodeFunctionData("updateTrackers", [voter]))
+  );
+  await tx.wait();
+  console.log("Done updating trackers for all voters");
+};
+
+export const getSumSlashedEvents = async (votingV2: VotingV2Ethers): Promise<BigNumber> => {
+  const voterSlashedEvents = await votingV2.queryFilter(votingV2.filters.VoterSlashed(), 0, "latest");
+  return voterSlashedEvents
+    .map((voterSlashedEvent) => voterSlashedEvent.args.slashedTokens)
+    .reduce((a, b) => a.add(b), ethers.BigNumber.from(0));
+};
+
+export const getNumberSlashedEvents = async (votingV2: VotingV2Ethers): Promise<number> => {
+  const voterSlashedEvents = await votingV2.queryFilter(votingV2.filters.VoterSlashed(), 0, "latest");
+  return voterSlashedEvents.length;
+};
+
+export const getVotingTokenExternalTransfersAmount = async (
+  votingToken: VotingTokenEthers,
+  votingV2: VotingV2Ethers
+): Promise<BigNumber> => {
+  const transferEvents = await votingToken.queryFilter(
+    votingToken.filters.Transfer(null, votingV2.address, null),
+    0,
+    "latest"
+  );
+
+  let sumExternalTransfers = BigNumber.from(0);
+
+  for (let i = 0; i < transferEvents.length; i++) {
+    const transferEvent = transferEvents[i];
+    const allVotingV2Events = await votingV2.queryFilter(
```

Comment:
I might be missing something here, but why not do this query over the entire block range outside of the loop? Is this to avoid hitting infura's 10,000 event limit?